api/incubator/src/test/java/io/opentelemetry/api/incubator/config/GlobalConfigProviderTest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.incubator.config;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.opentelemetry.api.incubator.config.internal.YamlStructuredConfigProperties;
+import java.util.Collections;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class GlobalConfigProviderTest {
+
+  @BeforeAll
+  static void beforeClass() {
+    GlobalConfigProvider.resetForTest();
+  }
+
+  @AfterEach
+  void after() {
+    GlobalConfigProvider.resetForTest();
+  }
+
+  @Test
+  void setAndGet() {
+    assertThat(GlobalConfigProvider.get()).isEqualTo(ConfigProvider.noop());
+    ConfigProvider configProvider =
+        () -> YamlStructuredConfigProperties.create(Collections.emptyMap());
+    GlobalConfigProvider.set(configProvider);
+    assertThat(GlobalConfigProvider.get()).isSameAs(configProvider);
+  }
+
+  @Test
+  void setThenSet() {
+    GlobalConfigProvider.set(() -> YamlStructuredConfigProperties.create(Collections.emptyMap()));
+    assertThatThrownBy(
+            () ->
+                GlobalConfigProvider.set(
+                    () -> YamlStructuredConfigProperties.create(Collections.emptyMap())))
```

Comment:
prefer pulling this out so that the assertion only has one variable. I know the actual assert has one sensible failure case, but cleaner imho